backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -81,27 +94,53 @@ public TestOrderService(
     return _repo.fetchQueue(fac.getOrganization(), fac);
   }
 
-  @Transactional(readOnly = true)
-  @AuthorizationConfiguration.RequirePermissionReadResultListAtFacility
-  public List<TestEvent> getTestEventsResults(UUID facilityId, int pageOffset, int pageSize) {
-    Facility fac = _os.getFacilityInCurrentOrg(facilityId); // org access is checked here
-    return _terepo.getTestEventResults(fac.getInternalId(), PageRequest.of(pageOffset, pageSize));
-  }
-
-  public int getTestResultsCount(UUID facilityId) {
-    Facility fac = _os.getFacilityInCurrentOrg(facilityId); // org access is checked here
-    return _terepo.getTestResultsCount(fac.getInternalId());
+  // Specifications filters for queries
+  private Specification<TestEvent> buildTestEventSearchFilter(
```

Comment:
Wow, this is some fancy stuff, I've never seen this `Specification` class before. Not totally sure I understand how it works, but I like it better than having a bunch of different methods in the Service, great work!